src/Compute/Compute/VirtualMachine/Operation/NewAzureVMCommand.cs

```diff
@@ -507,42 +510,65 @@ public void DefaultExecuteCmdlet()
                         }
                     }
 
-                    var result = this.VirtualMachineClient.CreateOrUpdateWithHttpMessagesAsync(
-                        this.ResourceGroupName,
-                        this.VM.Name,
-                        parameters,
-                        auxAuthHeader).GetAwaiter().GetResult();
-                    var psResult = ComputeAutoMapperProfile.Mapper.Map<PSAzureOperationResponse>(result);
-
-                    if (!(this.DisableBginfoExtension.IsPresent || IsLinuxOs()))
+                    if (NoWait.IsPresent)
                     {
-                        var currentBginfoVersion = GetBginfoExtension();
+                        var result = this.VirtualMachineClient.BeginCreateOrUpdateWithHttpMessagesAsync(
+                            this.ResourceGroupName,
+                            this.VM.Name,
+                            parameters,
+                            auxAuthHeader);
+                        var psResult = ComputeAutoMapperProfile.Mapper.Map<PSAzureOperationResponse>(result);
+
+                        if (!(this.DisableBginfoExtension.IsPresent || IsLinuxOs()))
+                        {
+                            var currentBginfoVersion = GetBginfoExtension();
 
-                        if (!string.IsNullOrEmpty(currentBginfoVersion))
+                            if (!string.IsNullOrEmpty(currentBginfoVersion))
+                            {
+                                WriteWarning("BGInfo extension not created in NoWait mode.");
+                            }
+                        }
+                        WriteObject(psResult);
+                    }
+                    else
```

Comment:
I am not certain the BGInfoExtension calls will work in this case, I would revert this one

Reply:
Done